### ISSUE TITLE
Request - support for inheritance in styles

### ISSUE THREAD

**Issue by reubenfirmin** [rjaros/kvision#300]
The upgrade to 5 made my workaround for https://github.com/rjaros/kvision/issues/258 no longer work, so I switched to the parentStyle syntax (since the prior issue is fixed). Unfortunately it doesn't seem to have any effect; i.e. only the immediate style seems to have effect.

**Comment by rjaros** [comment]
Style objects are developed to recreate CSS cascading styles. And specifying parent object doesn't make the child inherit parent
properties. It's about nesting declarations. It just renders the correct selectors in a hierarchical CSS structure.

To inherit properties from other style object in KVision 5 you can use this extension function:
```
fun Style.inheritFrom(other: Style): Style {
    val otherStyles = other.getSnStyle()
    for (key in js("Object").keys(otherStyles)) {
        setStyle(key, otherStyles[key])
    }
    return this
}

```

**Comment by reubenfirmin** [comment]
Got it. May I turn this into a request then for OO-style inheritance to be supported in KVision? In the app I'm building I have "cards" that all share a common style, and then go through a hierarchy of different styles depending on the state that they're representing.

**Comment by rjaros** [comment]
Released with 5.2.0

**Comment by reubenfirmin** [comment]
Verified - this works great. Thanks
